src/OrchardCore.Modules/OrchardCore.Users/ViewModels/LinkExternalLoginViewModel.cs

```diff
@@ -0,0 +1,23 @@
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using OrchardCore.Users.Controllers;
+
+namespace OrchardCore.Users.ViewModels
+{
+    public class LinkExternalLoginViewModel : IValidatableObject
+    {
+        [DataType(DataType.Password)]
+        public string Password { get; set; }
+
+        public IEnumerable<ValidationResult> Validate(ValidationContext validationContext)
+        {
+            var S = validationContext.GetService<IStringLocalizer<LinkExternalLoginViewModel>>();
```

Comment:
I think using DI is better, so we can use the localizer everywhere in the class

Reply:
I agree, but then we should register the view models with DI and inject them in constructors.

Reply:
> but then we should register the view models with DI 

Why? What I want to inject the `IStringLocalizer` in the `LinkExternalLoginViewModel`

Reply:
How would you inject the IStringLocalizer to the LinkExternalLoginViewModel without registering LinkExternalLoginViewModel in the container and request a resource of type IStringLocalizer in it's constructor ?

Reply:
Ok, just leave my previous comment